Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -408,50 +438,22 @@ For apps that handle sensitive data, make sure that the refresh token expires af
-
-Failing to destroy the server-side session is one of the most common logout functionality implementation errors . This error keeps the session or token alive, even after the user logs out of the application. An attacker who gets valid authentication information can continue to use it and hijack a user account.
-
-Many mobile apps don't automatically log users out because it is inconvenient for customers by implementing stateless authentication. The application should still have a logout function, and it should be implemented according to best practices, destroying the access and refresh token on the client and server. Otherwise, authentication can be bypassed when the refresh token is not invalidated.
-
-#### Verifying Best Practices
-
-If server code is available, make sure logout functionality terminates the session is terminated . This verification will depend on the technology. Here are examples session termination for proper server-side logout:
-
-- [Spring (Java)](https://docs.spring.io/autorepo/docs/spring-security/4.1.x/apidocs/org/springframework/security/web/authentication/logout/SecurityContextLogoutHandler.html "Spring (Java)")
-- [Ruby on Rails](https://guides.rubyonrails.org/security.html "Ruby on Rails")
-- [PHP](https://php.net/manual/en/function.session-destroy.php "PHP")
-
-If access and refresh tokens are used with stateless authentication, they should be deleted from the mobile device. The [refresh token should be invalidated on the server](https://auth0.com/blog/blacklist-json-web-token-api-keys/ "Blacklisting JSON Web Token API Keys").
-
-#### Dynamic Analysis
-
-Use an interception proxy for dynamic application analysis. Use the following steps to check whether the logout is implemented properly.
-
-1. Log into the application.
-2. Perform a couple of operations that require authentication inside the application.
-3. Log out.
-4. Resend one of the operations from step 2 with an interception proxy (Burp Repeater, for example). . This will send to the server a request with the session ID or token that was invalidated in step 3.
-
-If logout is correctly implemented on the server, an error message or redirect to the login page will be sent back to the client. On the other hand, if you receive the same response you got in step 2, the token or session ID is still valid and hasn't been correctly terminated on the server.
-The OWASP Web Testing Guide ([OTG-SESS-006](https://www.owasp.org/index.php/Testing_for_logout_functionality_%28OTG-SESS-006%29 "OTG-SESS-006")) includes a detailed explanation and more test cases.
+Also, make sure to check out the [OWASP JWT Cheat Sheet](https://goo.gl/TGzA5z "JSON Web Token (JWT) Cheat Sheet for Java") for additional information.
 
 ### Testing OAuth 2.0 Flows
```

Comment:
shouldn't we add a MSTG ID here as well as in the rest of the sections on the same markdown level?

Reply:
Agree. I added MSTG‑AUTH‑1 and MSTG‑AUTH‑3